platform/platform_windows.go

```
@@ -95,6 +96,62 @@ func netServerGetInfo() (si SERVER_INFO_101, err error) {
 	return platGetServerInfo(outdata), nil
 }
 
+func fetchOsDescription() (string, error) {
+	err := winbrand.Load()
+	if err == nil {
+		procBrandingFormatString := winbrand.NewProc("BrandingFormatString")
+		if procBrandingFormatString.Find() == nil {
+			// Encode the string "%WINDOWS_LONG%" to UTF-16 and append a null byte for the Windows API
+			magicString := utf16.Encode([]rune("%WINDOWS_LONG%" + "\x00"))
+			os, _, err := procBrandingFormatString.Call(uintptr(unsafe.Pointer(&magicString[0])))
+			defer syscall.LocalFree(syscall.Handle(os))
+			if err == ERROR_SUCESS {
+				return windows.UTF16PtrToString((*uint16)(unsafe.Pointer(os))), nil
+			}
+		}
+	}
+
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE,
+		registryHive,
+		registry.QUERY_VALUE)
+	defer k.Close()
+	if err == nil {
+		os , _, _ := k.GetStringValue(productNameKey)
+		return os, nil
+	}
+	
+	return "(undetermined windows version)", err
+}
+
+func fetchWindowsVersion() (uint64,uint64,uint64,error) {
+	var err error
+	var major uint64
+	var minor uint64
+	var build uint64
+
+	var osversion OSVERSIONINFOEXW
+	status, _, _ := procRtlGetVersion.Call(uintptr(unsafe.Pointer(&osversion)))
+	if status == 0 {
+		major = uint64(osversion.dwMajorVersion)
+		minor = uint64(osversion.dwMinorVersion)
+		build = uint64(osversion.dwBuildNumber)
+	} else {
+		k, err := registry.OpenKey(registry.LOCAL_MACHINE,
+			registryHive,
+			registry.QUERY_VALUE)
+		defer k.Close()
+		if err != nil {
+			var regmajor, _, _ = k.GetIntegerValue(majorKey)
+			var regminor, _, _ = k.GetIntegerValue(minorKey)
+			var regbuild, _, _ = k.GetStringValue(buildNumberKey)
+			major = regmajor
+			minor = regminor
+			build, _ = strconv.ParseUint(regbuild, 10, 0)
```

We should not ignore the error here